Sources/Sentry/SentryEnvelope.m

```diff
@@ -198,6 +206,38 @@ - (_Nullable instancetype)initWithAttachment:(SentryAttachment *)attachment
     return [self initWithHeader:itemHeader data:data];
 }
 
+- (nullable instancetype)initWithReplayEvent:(SentryReplayEvent *)replayEvent
+                             replayRecording:(SentryReplayRecording *)replayRecording
+                                       video:(NSURL *)videoURL
+{
+    NSData *replayEventData = [SentrySerialization dataWithJSONObject:[replayEvent serialize]];
+    NSData *recording = [SentrySerialization dataWithReplayRecording:replayRecording];
+    NSURL *envelopeContentUrl =
+        [[videoURL URLByDeletingPathExtension] URLByAppendingPathExtension:@"dat"];
+
+    BOOL success = [SentryMsgPackSerializer serializeDictionaryToMessagePack:@{
+        @"replay_event" : replayEventData,
+        @"replay_recording" : recording,
+        @"replay_video" : videoURL
+    }
+                                                                    intoFile:envelopeContentUrl];
+    if (success == NO) {
+        SENTRY_LOG_DEBUG(@"Could not create MessagePack for session replay envelope item.");
```

Comment:
`l`: I think this could be an error.
```suggestion
        SENTRY_LOG_ERROR(@"Could not create MessagePack for session replay envelope item.");
```